### PR TITLE
Compatibility issues with current rust and fortran versions resolved

### PR DESCRIPTION
**Pull request by HenrikJStromberg**
When trying to use the crate openBLAS failed to link with gfortran on all my systems. I diagnosed the issue to be caused by use of an outdated version of openblas-src. After updating I found, that superlu-sys uses mem::uninitialized in its test which is depricated. I swaped it for MaybeUninit. After these changes cargo test runs on all systems I have tested on.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,11 +2,10 @@
 name = "superlu-sys"
 version = "0.3.2"
 license = "Apache-2.0/MIT"
-authors = ["Ivan Ukhov <ivan.ukhov@gmail.com>"]
-description = "The package provides bindings to SuperLU."
+authors = ["Ivan Ukhov <ivan.ukhov@gmail.com>", "Henrik Stromberg <henrik@askdrq.com>"]
```
Comment (IvanUkhov) [inline Cargo.toml:5]: Please let us have one line per author and sorted alphabetically.

File: Cargo.toml [hunk 2]
```diff
@@ -2,11 +2,10 @@
...
+description = "The package provides bindings to SuoerLU. Fork of superlu-sys by Ivan Ukov."
```
Comment (IvanUkhov) [inline Cargo.toml:6]: You can undo this change.

File: Cargo.toml [hunk 3]
```diff
@@ -2,11 +2,10 @@
...
 documentation = "https://docs.rs/superlu-sys"
-homepage = "https://github.com/stainless-steel/superlu-sys"
-repository = "https://github.com/stainless-steel/superlu-sys"
+repository = "https://github.com/HenrikJStromberg/superlu-sys"
```
Comment (IvanUkhov) [inline Cargo.toml:8]: And this.

File: README.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 # superlu-sys [![Package][package-img]][package-url] [![Documentation][documentation-img]][documentation-url] [![Build][build-img]][build-url]
 
 The package provides bindings to [SuperLU].
+If is a fork of superlu-sys by Ivan Ukov.
```
Comment (IvanUkhov) [inline README.md:4]: And this.

File: src/supermatrix.rs [hunk 5]
```diff
@@ -49,6 +50,37 @@ pub struct SuperMatrix {
     pub Store: *mut c_void,
 }
 
+impl SuperMatrix {
+    pub fn data_as_vec(&self) -> Option<Vec<f64>> {
```
Comment (IvanUkhov) [inline src/supermatrix.rs:54]: I think the idiomatic name would be `to_vec`. `as_` is more for borrowing.

File: src/supermatrix.rs [hunk 6]
```diff
@@ -49,6 +50,37 @@ pub struct SuperMatrix {
...
+        match self.Stype {
+            Stype_t::SLU_DN => {}
+            _ => {return None}
```
Comment (IvanUkhov) [inline src/supermatrix.rs:57]: Please run `cargo fmt`.

**Review by IvanUkhov — COMMENTED** [review]
Thank you for the update!

Would you perhaps be interested in becoming a maintainer/owner for this repository and the corresponding crate? I can add you everywhere so that you can push and publish on your own.

**Comment by HenrikJStromberg** [comment]
That would be great as I will probably do more work on the crate in the future as a project of mine depends on it. Do you want to keep the superlu crate as is or should I also open merge requests there to merge my high level abstraction [sprs-spuerlu](https://github.com/HenrikJStromberg/sprs-superlu) into it?

**Comment by IvanUkhov** [comment]
Let us start with `superlu-sys`, and regarding `superlu`, I would propose to keep it as is for now. It is supposed to be a slim wrapper with native Rust types. It was not meant to have any higher-level functions built on top of the low-level ones, which is what `sprs-spuerlu` is doing, if I understand correctly. Even the function you just added is already a little bit misplaced; this crate was meant to have only bindings.

**Comment by IvanUkhov** [comment]
I have added you on crates.io as well. So you can publish.

**Comment by IvanUkhov** [comment]
Added you to `superlu` as well.

Please remember to undo your fork-related changes. They are now on master.

**Comment by HenrikJStromberg** [comment]
Thanks for the reminder. According to your which I have uploaded a new version of superlu-sys.
I am unsure about your wishes concerning sprs-superlu, as you have added me to the superlu repo but said you were consider sprs-superlu to go in a different direction then your superlu crate. shell I merge with superlu or make my own?

**Comment by IvanUkhov** [comment]
Sorry for the confusion. I thought it would not hurt to give you access to that one. You based your `sprs-superlu` on `superlu-sys`, but the intention was that people would be using `superlu`. Since you didn't go that way, perhaps there is something that would have to change there for it to become usable for you. Ideally, your high-level functions in `sprs-superlu` should depend on `superlu`, not `superlu-sys`.
